lib/types/monday-api-client.ts

```diff
@@ -0,0 +1,60 @@
+
+export type APIOptions = {
```

Comment:
IMO we don't need it, as it allows the developer to use different tokens and apiVersions for different api calls using the same client. This is not standard in the industry. Regarding the variables param - simple move it to the api function and maybe change its type to `Record<string, any>`

Reply:
This may make the life of developers a little bit harder. If they have one mutation that they don't know how to deprecate yet (move to the new version) and they know how to move the rest of their stuff - Without this option, They will "have" to stay on the old version instead of moving and staying in the old version only for this mutation

Reply:
@DorShakedMonday in that case they can simply create another ApiClient with the right version and use it for their deprecated methods